client/app/components/viz/VariantInspectCard.vue

```diff
@@ -2434,6 +2448,16 @@ export default {
           self.loadData();
       })
     },
+
+    mosaicVariant: {
+      handler(newMosaicVariant, oldMosaicVariant) {
+        if (newMosaicVariant !== oldMosaicVariant) {
+          this.updateMosaicVariantObject(newMosaicVariant);
+        }
+      },
+      deep: true,
```

Comment:
Why do we need a deep comparison here? Does Mosaic variant have values that have objects? This deep watch can be expensive, so only use it if necessary. It seems like the conventional watch on Mosaic variant should suffice.

Reply:
I put deep watch here because there is array values in mosaicVariant object. The conventional watch is a shadow watch, according to the documentation, if the watch property is an object or array, the deep watch is recommended. Another reason of adding deep watch, this mosaicVariant object is passed down to child component as a prop. But you're right, there is performance concern when using deep watch. I am not sure when part of data changes, the handler function will still be triggered without deep watch.